### DISCUSSION TITLE
skynet 热更新

### DISCUSSION THREAD

**Discussion by hujiangang** [cloudwu/skynet#1658] [Q&A]
新项目考虑用skynet,最近一直在边写边学习，关于热更新有一个疑虑
1.我需要更新同类型的服务代码，应该怎么做？ 不会一个一个 Inject address xx.lua 吧（类似agent服务很多个）。
2. clearcache 是清理全部已加载过的lua代码，那单个有没有？

**Comment by cloudwu** [comment]
1. skynet 没有提供热更新方案，留给你自己做。
2. Lua 的代码也是数据。只要你理解了 Lua 怎么工作，你能想出无数种方法做这个事情。每种方法必有取舍。
3. 如果你选择一个个发消息更新，我认为没有问题。应该也不存在什么魔法，可以冻结所有正在运行的服务，咔嚓一下全换过来，并保证有正确的行为。如果你实现了，可以分享出来。
4. clearcache 不是你想象的用途，它不是给热更新用的。
